fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalScanNode.java

```diff
@@ -90,4 +97,20 @@ public boolean needToCheckColumnPriv() {
     public int getNumInstances() {
         return scanRangeLocations.size();
     }
+
+    public void setNereidsConjuncts(Set<Expression> conjuncts) {
+        this.nereidsConjuncts = conjuncts;
+    }
+
+    public void setContext(PlanTranslatorContext context) {
+        this.context = context;
+    }
+
+    private void translateConjuncts() {
+        if (nereidsConjuncts != null) {
+            nereidsConjuncts.stream()
+                .map(e -> ExpressionTranslator.translate(e, context))
+                    .forEach(conjuncts::add);
+        }
```

Comment:
u should tranlate in PhysicalPlanTranslator and set legacy planner's expr（List<Expr>）to this node

Reply:
Moved the translate code to PhysicalPlanTranslator